### ISSUE TITLE
A single class for storing iSEE configuration ?

### ISSUE THREAD

**Issue by kevinrue** [iSEE/iSEE#314]
How about wrapping all the settings DataFrame's into a single object of class e.g. `iSEEParam`?

Some pros and cons:
1. (+) the call to `iSEE()` could be more compact if we separate the configuration of individual panels from more general app arguments
2. (+) we could give users getters and setters more convenient than the current `$` and nested `list` tricks to set values. Setters would also allow us to enforce validity checks to further help users avoid invalid values.
3. (-) Maintenance? I would really like to avoid the need for `updateObject` or users saving the configuration object as RDS. We need to make sure users are encouraged to continue saving their configuration as R scripts.

Concept code:

* Separate definition of iSEE parameters:

```
isp <- iSEEParams(
  redDimArgs = NULL, colDataArgs = NULL, featAssayArgs = NULL,
  rowStatArgs = NULL, rowDataArgs = NULL, sampAssayArgs = NULL,
  colStatArgs = NULL, customDataArgs = NULL, customStatArgs = NULL, heatMapArgs = NULL,
  redDimMax = 5, colDataMax = 5,
  featAssayMax = 5, rowStatMax = 5, rowDataMax = 5,
  sampAssayMax = 5, colStatMax = 5, customDataMax = 5,
  customStatMax = 5, heatMapMax = 5,
  initialPanels=NULL)
```

which - in our current examples - would generally be as short as

```
isp <- iSEEParams(redDimArgs = redDimArgs, initialPanels = initialPanels)
```

* Shorter call to `iSEE()`

```
iSEE <- function(se,
    ISEEParam = isp,
    annotFun=NULL, customDataFun=NULL, customStatFun=NULL, customSendAll=FALSE,
    colormap=ExperimentColorMap(), tour=NULL, appTitle=NULL, runLocal=TRUE,
    voice=FALSE)
```

* Getters / setters

```
redDimPlotArgs(isp) <- DataFrame(...)
redDimPlotArg(isp, index=1, field="XAxis") <- 3L
colDataPlotArg(isp, index=2, field="ColorBy") <- "Column data"
```

**Comment by federicomarini** [comment]
Reprising my own answer on the Slack channel.

This would also offer a solution to keep backwards compatibility. If we do something like the call above:

- we have the `iSEEparam` parameter which is a list with all the params, and sets these first
- if single parameters are specified as well, we keep these to be overridden? In a way that what is not specified in the iSp gets the default, and what is also explicitly defined as param, that gets the value one specified.

This would be a substantial help in going towards “mode- objects” for using iSEE - and hey, one can even share these in case a nice setup has been found.

Waiting for others to chip in, in case more thoughts need to be spent here.

**Comment by LTLA** [comment]
Should be closed by #327, where we can just pass the SE + list of objects pretty easily.
